tools/generator-bot-template-generator/app/index.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+const path = require('path');
+const Generator = require('yeoman-generator');
+const askName = require('inquirer-npm-name');
+const _ = require('lodash');
+const extend = require('deep-extend');
+const mkdirp = require('mkdirp');
+
+function makeGeneratorName(name) {
+  name = _.kebabCase(name);
+  name = name.indexOf('generator-') === 0 ? name : 'generator-' + name;
+  return name;
+}
+
+module.exports = class extends Generator {
+  initializing() {
+    this.props = {};
+  }
+
+  prompting() {
+    return askName(
+      {
+        name: 'name',
+        message: 'Your generator name',
+        filter: makeGeneratorName,
+        validate: str => {
+          return str.length > 'generator-'.length;
```

Comment:
Why does it need to be longer than 'generator-'? I'm not understanding the purpose of this at face value.

Reply:
For example, the user can enter either enter 'generator-conversational-core' or 'conversational-core', if they enter the ladder then the `filter` function will append 'generator-' as a prefix to 'conversational-core'. At the end of the day it needs that prefix and this allows the user to manually add it or not. 

The validate function just loosely ensures that 'generator-' is in the name by checking length.  its unlikely that this validate will ever be called and return false unless someone changes the `filter` function. I can remove the validate function entirely  for this reason or if desired to a stronger check to check that the name starts with 'generator-' as opposed to checking length.